contracts/core/mpl-rewards/v1/test/MplRewards.t.sol

```diff
@@ -1,157 +1,573 @@
-        assertEq(mplRewards.totalSupply(),                                 0);
-
-        assertTrue(!fakeGov.try_recoverERC20(USDC, 400 * USD));
-        assertTrue(     gov.try_recoverERC20(USDC, 400 * USD));
+contract MplRewardsTest is DSTest {
 
-        assertEq(IERC20(USDC).balanceOf(address(fay)),                     0);
-        assertEq(IERC20(USDC).balanceOf(address(gov)),             400 * USD);
-        assertEq(IERC20(USDC).balanceOf(address(mplRewards)),      600 * USD);
-        assertEq(mplRewards.balanceOf(address(fay)),                       0);
-        assertEq(mplRewards.totalSupply(),                                 0);
+    Hevm hevm;
 
-        assertTrue(!fakeGov.try_recoverERC20(USDC, 600 * USD));
-        assertTrue(     gov.try_recoverERC20(USDC, 600 * USD));
+    uint256 constant WAD = 10 ** 18;
 
-        assertEq(IERC20(USDC).balanceOf(address(fay)),                     0);
-        assertEq(IERC20(USDC).balanceOf(address(gov)),            1000 * USD);
-        assertEq(IERC20(USDC).balanceOf(address(mplRewards)),              0);
-        assertEq(mplRewards.balanceOf(address(fay)),                       0);
-        assertEq(mplRewards.totalSupply(),                                 0);
+    constructor() public {
+        hevm = Hevm(address(bytes20(uint160(uint256(keccak256("hevm cheat code"))))));
     }
 
-    function test_setRewardsDuration() public {
-        assertEq(mplRewards.periodFinish(),         0);
-        assertEq(mplRewards.rewardsDuration(), 7 days);
-
-        mpl.transfer(address(mplRewards), 25_000 * WAD);
-
-        gov.notifyRewardAmount(25_000 * WAD);
-
-        assertEq(mplRewards.periodFinish(),    block.timestamp + 7 days);
-        assertEq(mplRewards.rewardsDuration(),                   7 days);
-
-        assertTrue(!fakeGov.try_setRewardsDuration(30 days));
-        assertTrue(    !gov.try_setRewardsDuration(30 days));  // Won't work because current rewards period hasn't ended
-
-        hevm.warp(mplRewards.periodFinish());
-
-        assertTrue(!gov.try_setRewardsDuration(30 days));  // Won't work because current rewards period hasn't ended
-
-        hevm.warp(mplRewards.periodFinish() + 1);
-
-        assertTrue(gov.try_setRewardsDuration(30 days));  // Works because current rewards period has ended
-
-        assertEq(mplRewards.rewardsDuration(), 30 days);
+    function test_transferOwnership() external {
+        SomeAccount account1 = new SomeAccount();
+        SomeAccount account2 = new SomeAccount();
+        MplRewards rewardsContract = new MplRewards(address(0), address(1), address(account1));
+        
+        assertEq(rewardsContract.owner(), address(account1));
+
+        {
+            (bool success,) = account2.tryCall(
+                address(rewardsContract),
+                abi.encodeWithSignature("transferOwnership(address)", address(account1))
+            );
+            assertTrue(!success);
```

Comment:
any reason to change the methodology of tests? Instead of a dedicated function call, we have dynamic function signature calling. I can see the problem in this kind of process if the functional signature get change then we have to change a lot of places but those will be easy as well

Reply:
I see what you're doing here @deluca-mike but tbh I much prefer the dedicated wrapper functions. They're easier to read and keep the test sizes down. Plus for the `call` functions we don't have to decode the params.